### ISSUE TITLE
[Feature Request]: Expected Checkin Date for accessories

### ISSUE THREAD

**Issue by floschoepfer** [grokability/snipe-it#12384]
### Is your feature request related to a problem? Please describe.

Currently, you can only set an expected checkin date for assets, but not for accessories.

### Describe the solution you'd like

I would like to have the same ability to set an expected checkin date on accessories, since we give some accessories only for a short timespan out.
![image](https://user-images.githubusercontent.com/112496896/213677294-f4509d34-94e0-472d-af25-b6ff9dd6b8c1.png)

**Comment by eugene-bce** [comment]
+1 We are assigning laptop chargers and even HDMI cables as assets because we can't put an expected return date. School environment where we loan chargers and cables.

**Comment by floschoepfer** [comment]
> We are assigning laptop chargers and even HDMI cables as assets because we can't put an expected return date.

I thought about that too, but decided against it because it would become too confusing. We currently do it in a way that we enter the date as a comment when checking out the accessory. However, we don't get a message by email or can sort/filter according to it...

**Comment by jrainone** [comment]
This is a feature that we would like to implement.  Not everything is an asset but we expect them returned to us and the end user should be reminded.  a few use cases on our end are listed below.

We have an event where the external affairs team is requesting speakers, microphones, XLR cables and extension cords.  XLR cables and Microphones are not unique, do not contain any barcodes and any item can be swapped with other exact items.  we expect after the event that they will return those items once they are done with them.  These are not components since they are independent of the original asset and can quantity can vary between none and several depending on the need.. 

A user arrived to work without their laptop charger, wireless mouse, ect, we loan out the accessory but expect it to be returned by end of day. again these are not assets but we still do not expect them never to return.  These items don't rise to the level of asset but still need to be tracked.

**Comment by IdaEdz** [comment]
According to the [API reference guide ](https://snipe-it.readme.io/reference/accessories), accessories do have API calls for Expected checkin date (expected_checkin).

So there should be a way to implement this?

I agree with previous comments about having bulk of items like chargers or HDMI-adapters that are handed out for loans during a shorter time frame.

**Comment by floschoepfer** [comment]
> According to the [API reference guide ](https://snipe-it.readme.io/reference/accessories), accessories do have API calls for Expected checkin date (expected_checkin).

Interesting...

@snipe why is this not implemented in the UI, when it is possible with the API? Or is the API reference wrong?

**Comment by eugene-bce** [comment]
@floschoepfer - Interestingly the BlueTally fork of SnipeIT has been able to implement this (along with checking out to a location rather than a user)
![image](https://github.com/snipe/snipe-it/assets/35854972/946d136b-a15a-452d-9819-7c04a2116cde)

**Comment by eugene-bce** [comment]
@floschoepfer They also have a really nice dashboard for returns. 

![image](https://github.com/snipe/snipe-it/assets/35854972/f16626d3-626a-4453-9ad5-68fe63d77efe)

**Comment by RolandWarburton** [comment]
+1  This feature would be most welcome!
Being able to checkout an accessory with an expected return date is just as important to me an expected return date for an asset.

**Comment by Duckfromearth** [comment]
Big +1 this is the biggest issue we currently face with snipe

**Comment by snipe** [comment]
> Interestingly the BlueTally fork of SnipeIT has been able to implement this (along with checking out to a location rather than a user)

I am not aware of Bluetally being a fork of Snipe-IT (and if they are, they are likely in violation of our license.)

@Duckfromearth Is there a particular reason you’re not just listing those accessories as assets?

**Comment by Duckfromearth** [comment]
I don't want to track them as assets because they are non-unique items that have large quantities.  I'm actually using snipe IT to manage all the equipment for my video production business.  Much of our equipment is IT equipment, but we also have lots of other items.  I

We have for example 30 sandbags, they are a bulk item so we don't want to track them individually. They are entirely interchangeable.  We might checkout 15 at a time, and having to figure out for example which sandbags have the correct ID's is a ton of work.   However, we do want to make sure that everything that goes out comes back.  We would also love if they could have email notifications just like assets - but just having the expected by date would be enough, because we could build our own notification system with N8N.

I know I am not your average snipe IT user, but there are a surprising amount of folks in my industry who are also using SNIPE, and I think it would benefit the larger user base as well.

**Comment by Sykasa** [comment]
Is this worked on in any form right now? This feature is making me want to move away from Snipe, since it's just not feasible to use assets for simple things like cables, keyboards, etc.

**Comment by gkwurst** [comment]
This should be an easily implemented change.  We need to have check-in notifications for loaned accessories like AC chargers.  Please prioritize this.

**Comment by snipe** [comment]
> This should be an easily implemented change.

It really isn't though, since generally speaking, if you care enough about a particular thing to need it checked in at a certain date, it is better represented as an asset, not an accessory.

Accessories themselves are not bespoke. It's not THIS keyboard, it's A keyboard. The system currently doesn't really care which keyboard was assigned, since, again, you generally want those to be assets, just that one was assigned, so a particular keyboard (or whatever) needing to be checked in at a particular date doesn't jibe with how it was designed.

**Comment by gkwurst** [comment]
> It really isn't though, since generally speaking, if you care enough about a particular thing to need it checked in at a certain date, it is better represented as an asset, not an accessory.
> 
> Accessories themselves are not bespoke. It's not THIS keyboard, it's A keyboard. The system currently doesn't really care which keyboard was assigned, since, again, you generally want those to be assets, just that one was assigned, so a particular keyboard (or whatever) needing to be checked in at a particular date doesn't jibe with how it was designed.

As you stated, these are accessories and not assets.  We don't track things like keyboards and AC adapters as assets, but we would like them back.  If someone borrows one of 20 spare AC adapters I don't care which one in particular, I'd just like a reminder to be set that Steve borrowed an AC adapter and we'd like it back.

Currently we put checkout info in the Notes field, but with no ability to send a reminder we've already had accessories slip through the cracks and be lost.  We can see all the users an accessory has been checked-out to, we just would like an expected check-in option.

I'm not familiar with the database schema here, but if the expected check-in date on an asset is tied to that individual asset could it be tied to the Checkout event with an accessory?  Really anything to provide us with a notification would be helpful.

**Comment by snipe** [comment]
I understand the ask, but since you can’t edit an individual accessory (since they’re not bespoke) we don’t really have a way for you to edit that expected checkin if it were to change, while the expected checkin of an asset can be edited by editing the asset itself. Adding this would require a new (potentially confusing) UI, new logging system (logging the attributes of the checkout itself), updated API endpoints, and checkout notifications, validation flow - not even counting the actual reporting options in the original ask. 

That’s not to say it isn’t possible, but calling it an “easily implemented” change is quite an overstatement.

**Comment by gkwurst** [comment]
> That’s not to say it isn’t possible, but calling it an “easily implemented” change is quite an overstatement.

Not that it would be any less potentially confusing, but you could in theory tie an accessory checkout notification event to the user assigned instead of the event itself and manage notifications there.  That would probably be less awful than tying it to the event, but opens up a different can of worms I'm sure.  Even doing that it's still complicated, you'd have to have fields for the expected-checkin tied to an accessory type, and what if the user checked-out multiple of the same accessory?  As you said, not as easy as it would seem.

My thought that it should be easy was the fact that someone stated the API already had an expected_checkin field for the accessories, but with them not being unique that wouldn't work properly since the field would apply to the accessory itself and not each individual instance.

**Comment by snipe** [comment]
@gkwurst I don't think that was correct though - we've never had that capability on accessories. Perhaps a bug in the documentation, or perhaps the OP was confusing the docs. https://snipe-it.readme.io/reference/accessories-checkout 



<blockquote><img src="https://cdn.readme.io/og-image/create?type=reference&title=%2Faccessories%2F%3Aid%2Fcheckout&projectTitle=Snipe-IT%20Documentation&description=Check%20an%20accessory%20out%20to%20a%20user&logoUrl=https%3A%2F%2Ffiles.readme.io%2Fe9d890e-small-standalone-transparent-400.png&color=%23892e88&variant=light" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/reference/accessories-checkout">/accessories/:id/checkout</a></strong></div><div>Check an accessory out to a user</div></blockquote>
